include/xtypes/TypeKind.hpp

```diff
@@ -52,7 +52,7 @@ enum class TypeKind
 
     ENUMERATION_TYPE = CONSTRUCTED_TYPE | ENUMERATED_TYPE | 0x0001,
     BITSET_TYPE      = CONSTRUCTED_TYPE | 0x0002, ///< Not supported
-    ALIAS_TYPE       = CONSTRUCTED_TYPE | 0x0003, ///< Not supported
+    ALIAS_TYPE       = CONSTRUCTED_TYPE | 0x0003, ///< Represents an AliasType
```

Comment:
Thinking later about it, it had more cons that pros in my opinion. Using solely a bit without altering the rest made DynamicType kind checking quite difficult for certain situations, because all the & operations performed to an AliasType throughout the code (for example, checking if some DynamicType is primitive using DynamicType::is_primitive_type()) would return True for the AliasType object; this led to AliasType objects pointed by DynamicType::Ptr not cloning themselves, for example. Maybe a shortcut method could be added to AliasType class to retrieve the original TypeKind::KIND?

Reply:
Ok... at this time I suppose that is something like `alias.content().kind()`. It is ok for me. No shortcut is needed.

Reply:
Great :+1: